addons/mail/static/src/components/file_uploader/file_uploader_tests.js

```diff
@@ -87,6 +88,49 @@ QUnit.test('no conflicts between file uploaders', async function (assert) {
     );
 });
 
+QUnit.test('handles pre-upload removals of attachments', async function (assert) {
+    assert.expect(1);
+
+    const attachmentUploadedPromise = makeTestPromise();
+    await this.start({
+        async mockFetch(resource, init) {
+            const res = this._super(...arguments);
```

Comment:
`_super` should not be called before your promise, because that means in regard to the (mock) server the action has already been done even if you block the result of the rpc. (other parallel rpc might already ready the "new" values, such as fetching the new file by luck even though you want to simulate is not uploaded yet).

It is safer to just bind `_super` before, and only call it at the end.